src/components/Sidebar.js

```diff
@@ -0,0 +1,77 @@
+import Header from './Header.js';
+import DocumentList from './DocumentList.js';
+import Footer from './Footer.js';
+import { request } from '../utils/api.js';
+import { push } from '../utils/router.js';
+
+export default function Sidebar({
+  $target,
+  initialState
+}){
+  const $sidebar = document.createElement('section');
+  $sidebar.className = 'sidebar';
+  
+  this.state = initialState;
+
+  this.setState = () => {
+    getDocumentList();
+    this.render();
+  }
+
+  new Header({
+    $target: $sidebar
+  })
+  
+  const documentList = new DocumentList({
+    $target: $sidebar,
+    initialState: [],
+    onSelect: (id) => {
+      push(`/documents/${id}`);
+    },
+    onCreateDocument: async (parentId) => {
+      const document = {
+        title: 'Untitled',
+        parent: parentId,
+      };
+      const newDocument = await createDocument(document);
+      push(`/documents/${newDocument.id}`);
+    },
+    onRemoveDocument: async (id) => {
+      await request(`/documents/${id}`, {
+        method: 'DELETE'
+      });
+      push('/');
+    }
+  });
+
+  new Footer({
+    $target: $sidebar,
+    onCreateRootDocument: async () => {
+      const document = {
+        title: 'Untitled',
+        parent: null,
+      };
+      const newDocument = await createDocument(document);
+  
+      push(`/documents/${newDocument.id}`);
+    }
+  });
+  
+  this.render = () => {
+    getDocumentList();
+    $target.prepend($sidebar);
+  };
```

Comment:
p3;
pr에서 언급하신 section 별 렌더링이 이부분이군요! Header와 Footer는 한 번만 렌더링 되면 되고, DocumentList 컴포넌트만 상태에 따라 렌더링하면 되기 때문에 아래와 비슷하게 해결 할 수도 있을 것 같습니다!

```suggestion
  let isInit = false

  this.render = () => {
    if (!this.isInit) {
      $target.appendChild($sidebar);
      isInit = true;
    }
  };
```